### PR TITLE
DOC: How to deprecate

### PR DESCRIPTION
**Pull request by MartinThoma**

### CODE REVIEW

**Comment by MartinThoma** [comment]
TODO: 

* Change "How to deprecate"to "Deprecation Process"
* Mention version pinning / semantic versioning

File: docs/dev/deprecations.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ones. We are careful with introducing potentially breaking changes, but we
+will do them if they provide value for the community on the long run.
+
+## How to Deprecate
+
+Assume the current version of PyPDF2 is `x.y.z`. After a discussion (e.g. via
+Github issues) we decided to remove a class / function / method. This is how
+we do it:
+
+1. `x.y.(z+1)`: Add a PendigDeprecationWarning. If there is a replacement,
```
Comment (MasterOdin) [inline docs/dev/deprecations.md:13]:
```suggestion
1. `x.y.(z+1)`: Add a PendingDeprecationWarning. If there is a replacement,
```

File: docs/dev/deprecations.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+   the replacement is also introduced and the warning informs about the change
+   and when it will happen.
+   The docs let users know about the deprecation and when it will happen and the new function.
+   The CHANGELOG informs about it.
+2. `(x+1).0.0`: The PendingDeprecationWarning is changed to a DeprecationWarning.
+   The CHANGELOG informs about it.
+3. `(x+2).0.0`: The code and the DeprecationWarnings are removed.
+   The CHANGELOG informs about it.
+
+This means the users have 3 warnings in the CHANGELOG, a PendigDeprecationWarning
```
Comment (MasterOdin) [inline docs/dev/deprecations.md:23]:
```suggestion
This means the users have 3 warnings in the CHANGELOG, a PendingDeprecationWarning
```

**Review by MasterOdin — COMMENTED** [review]
I'm neutral on the change as I don't think deprecations should happen often enough to have a formal documented setup, but can't hurt I suppose to have.

**Pull request merged**